lib/scala/project-manager/src/main/scala/org/enso/projectmanager/service/ProjectService.scala

```diff
@@ -490,20 +517,17 @@ class ProjectService[
       )
     }
 
-    projectTemplate match {
-      case Some(_) =>
-        CovariantFlatMap[F]
-          .ifM(projectRepository.exists(projectName))(
-            ifTrue  = findAvailableName(projectName, 1),
-            ifFalse = CovariantFlatMap[F].pure(projectName)
-          )
-          .mapError(toServiceFailure)
-      case None =>
-        CovariantFlatMap[F].pure(projectName)
-    }
-
+    CovariantFlatMap[F]
+      .ifM(projectRepository.exists(projectName))(
+        ifTrue  = findAvailableName(projectName, 1),
+        ifFalse = CovariantFlatMap[F].pure(projectName)
+      )
+      .mapError(toServiceFailure)
   }
 
+  private def getNameForDuplicatedProject(projectName: String): String =
```

Comment:
What if there is already one that exists?

Reply:
This function returns the suggested project name, and then we use `getNameForNewProject` function that checks if the current name exists and returns the final name for the project.